packages/frama-clang/frama-clang.0.0.13/opam

```diff
@@ -0,0 +1,46 @@
+opam-version: "2.0"
+synopsis: "Frama-Clang, Frama-C's C++ front-end"
+description:"""
+Frama-Clang is a plug-in of Frama-C that allows parsing C++ files.
+It is based on clang/LLVM
+"""
+maintainer: "virgile.prevosto@cea.fr"
+authors: [
+  "David Cok"
+  "Virgile Prevosto"
+  "Franck Védrine"
+]
+homepage: "https://frama-c.com/frama-clang.html"
+license: "LGPL-2.1-only"
+dev-repo: "git+https://git.frama-c.com/pub/frama-clang.git"
+bug-reports: "https://git.frama-c.com/pub/frama-clang/-/issues"
+tags: [
+  "Frama-C"
+  "C++"
+]
+
+url {
+  src: "https://frama-c.com/download/frama-clang-0.0.13.tar.gz"
+  checksum: [ "sha512=d4fbfc170b43e7486fd6920363fe5ed73cfd6e1d537e6f64790f5cf9384e6d0c73eb626ee8c79d5492ffc559eec8ad649ed6ed2e5caedb519ab74c781f34f6ff" ]
+}
+
+build: [
+  ["sed" "-i" "-e" "s|llvm-config|%{conf-llvm:config}%|g" "configure"]
+  ["./configure"]
+  [make "-j%{jobs}%"]
+]
+
+install: [
+  [make "install"]
+]
+
+depends: [
+  "ocaml" { >= "4.08.1" }
+  "frama-c" { >= "25.0" & < "26.0~" }
+  "camlp5"
+  "camlp-streams"
+  "conf-llvm" { >= "10.0.0" }
+  "conf-libclang" { >= "10" }
```

Comment:
older versions of this package is not consistent. For example on debian stable, `conf-libclang.15` will install `libclang-dev` which is libclang 11 instead of 15. I'm not sure how much of a problem it is but it's worth mentioning.

Reply:
Actually, the main point is to have the same version than the llvm library, but  as far as I can tell from reading the opam manual, this is not really possible to express with package formulas

Reply:
It is possible (see the conf-llvm package) but rather annoying to maintain

Reply:
Sorry I meant that ideally the package formula should check that `conf-llvm` and `conf-libclang` share the same major version number. But you're right, the `depexts` in the various `conf-libclang.*/opam` are a bit odd. Furthermore, the `configure.sh` of `conf-libclang` is really testing the C bindings of clang. Maybe the suggestion of @raphael-proust to create a `conf-libclang-cpp` is the way to go. I'll give it a try.